docs/how-to/create-slice.rst

```diff
@@ -0,0 +1,107 @@
+How-to create a package slice
+******************************
+
+Problematic
+-----------------
+
+Are you tired of having an image weighing several MB for a software, or an image with many security vulnerabilities? In this tutorial we will show you how to avoid this with slices. In this tutorial we will create a slice name bins to only have the openssl binary and c_rehash in our image.
+
+Structure of a slice
+--------------------
+
+Here the structure of a slice : 
+
+.. code-block:: yaml
+
+    <name_of_the_slice>
+        <essential>
+        <contents>
+        <mutate>
+
+| The essential part is optional and permits you to install the dependencies of the packages you want to slice.
+| The contents part is required and permits you to keep the files from the package you want to keep in the slice.
+| The mutate part is optional and allows the reproduction of maintainer scripts. (use this https://github.com/google/starlark-go)
+
+Find dependencies of our packages
+---------------------------------
+
+So, the first step is to find the dependencies of the openssl package. We can easily do it with this command: 
+
+.. code-block:: sh
+
+    $ apt show openssl
+
+.. code-block:: sh
+
+    package: openssl
+    Version: 3.0.2-0ubuntu1.7
+    Origin: Ubuntu
+    Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
+    Original-Maintainer: Debian OpenSSL Team <pkg-openssl-devel@alioth-lists.debian.net>
+    Bugs: https://bugs.launchpad.net/ubuntu/+filebug
+    Depends: libc6 (>= 2.34), libssl3 (>= 3.0.2-0ubuntu1.2)
+
+So we see that openssl depends on libc6 and libssl3 (these slices already exist; we will see them after)
+
+Creating our slice
+------------------
+
+Now we will create our openssl slice, but first, you need to clone this repository (our slice database):
```

Comment:
```suggestion
You now have everything needed to actually define the OpenSSL slice that will install the content you're looking to have in the final ROCK.

Since you're looking to install just the OpenSSL binaries from the ``openssl`` package, what about naming this new slice _**bins**_? Let's go for it:

1. **What is the name of your slice definitions file?** It is a YAML file called ``openssl.yaml``
2. **What package name should be defined inside this file?** The package name is ``openssl``
3. **What is your slice name?** It should be called ``bins``
4. **What contents do you need from the OpenSSL package?** Just the binaries - ``/usr/bin/c_rehash`` and ``/usr/bin/openssl``
5. **Does your slice depend on any other package slice?** Yes, OpenSSL depends on ``libc6`` and ``libssl3``
    - **Do these two packages have slice definitions files upstream?** Yes, there's already a slice definitions file for [libc6](https://github.com/canonical/chisel-releases/blob/ubuntu-22.04/slices/libc6.yaml) and another for [libssl3](https://github.com/canonical/chisel-releases/blob/ubuntu-22.04/slices/libssl3.yaml)
        - If these dependencies were not present in the upstream Chisel release, you'd also need to create their corresponding slice definitions
    - **Which slices do you depend on then?** Since you only want the OpenSSL binaries, you might only need the libraries from ``libc6`` and ``libssl3``, as well as the configuration files from ``libc6`` and ``openssl`` itself.
    
With all this figured out, you can now create a new YAML file named ``openssl.yaml``, with the following content:

.. code-block:: yaml
    package: openssl
    slices:
        bins:
            essential:
            - libc6_libs
            - libc6_config
            - libssl3_libs
            - openssl_config
            contents:
              /usr/bin/c_rehash:
              /usr/bin/openssl:
            
        config:
            contents:
              /etc/ssl/private/:
              /etc/ssl/openssl.cnf:
              /usr/lib/ssl/certs:
              /usr/lib/ssl/openssl.cnf:
              /usr/lib/ssl/private:

Notice the unforeseen new slice ``config``. Because your OpenSSL binaries depend on the OpenSSL configuration files, and those weren't yet present anywhere in the Chisel releases upstream, you also need to create that slice! You may also ask _"why not put those configuration files inside the ``bins`` slice"_? You could! But we recommend, as a best practice, to separate and group contents according to their nature, as you may tomorrow need to create a new slice definition that only needs the OpenSSL configurations and not the binaries.

---

And that's it. This is your brand new slice definitions file, which will allow Chisel to install **just** the OpenSSL binaries (and their dependencies) into your ROCK! To learn about how to actually use this new slice definition file and publish it upstream for others to use, please check the following guides.
        
```